txstratum/healthcheck.py

```diff
@@ -0,0 +1,265 @@
+            setattr(self, key, value)
+
+    def to_json(self) -> Dict[str, str]:
+        """Return a dict representation of the object. All field names are converted to camel case."""
+        json = {
+            "componentType": self.component_type.value,
+            "status": self.status.value,
+            "output": self.output,
+        }
+
+        if self.time:
+            json["time"] = self.time
+
+        if self.component_id:
+            json["componentId"] = self.component_id
+
+        if self.observed_value:
+            assert (
+                self.observed_unit is not None
+            ), "observed_unit must be set if observed_value is set"
+
+            json["observedValue"] = self.observed_value
+            json["observedUnit"] = self.observed_unit
+
+        return json
+
+
+@dataclass
+class HealthCheckResult:
+    """This class is used to store the result of a health check in the tx-mining-service."""
+
+    status: HealthCheckStatus
+    description: str
+    checks: Dict[str, List[ComponentHealthCheck]]
+
+    def get_http_status_code(self) -> int:
+        """Return the HTTP status code for the status."""
+        if self.status == HealthCheckStatus.PASS:
+            return 200
+        elif self.status == HealthCheckStatus.WARN:
+            return 503
+        elif self.status == HealthCheckStatus.FAIL:
+            return 503
+        else:
+            raise ValueError("Invalid status")
+
+    def to_json(self) -> Dict[str, Any]:
+        """Return a dict representation of the object. All field names are converted to camel case."""
+        return {
+            "status": self.status.value,
+            "description": self.description,
+            "checks": {k: [c.to_json() for c in v] for k, v in self.checks.items()},
+        }
+
+
+class ComponentHealthCheckInterface(ABC):
+    """This is an interface to be used by other classes implementing health checks for components."""
+
+    @abstractmethod
+    async def get_health_check(self) -> ComponentHealthCheck:
+        """Return the health check status for the component."""
+        raise NotImplementedError()
+
+
+class HealthCheck:
+    """This is the main class that will use the other classes to check the health of the components.
+
+    It will aggregate the responses into a final object to be returned following our standards.
+    """
+
+    def __init__(self, manager: "TxMiningManager", backend: "HathorClient") -> None:
+        """Init the class with the components that will be checked."""
+        self.health_check_components: List[ComponentHealthCheckInterface] = [
+            MiningHealthCheck(manager),
+            FullnodeHealthCheck(backend),
+        ]
+
+    async def get_health_check(self) -> HealthCheckResult:
+        """Return the health check status for the tx-mining-service."""
+        components_health_checks = [
+            await c.get_health_check() for c in self.health_check_components
+        ]
```

Comment:
You could use [`asyncio.gather()`](https://docs.python.org/3/library/asyncio-task.html#asyncio.gather) here to run all futures concurrently. Probably won't change much in practice, but it's nice

Reply:
Done in https://github.com/HathorNetwork/tx-mining-service/pull/104/commits/d7880ffb00149c1584a54c4effae123ecd2e6d85